aten/src/ATen/native/LinearAlgebraUtils.h

```diff
@@ -48,6 +75,78 @@ static inline int64_t matrixStride(const Tensor& batched_matrices) {
   return batched_matrices.size(-1) * batched_matrices.size(-2);
 }
 
+// we assume that `a` and `b` do not overlap, and moreover,
+// we expect that there exist tensors a_c and b_c such that
+// a = a_c.contiguous().transpose(-1, -2)
+// b = b_c.contiguous().transpose(-1, -2)
+template<typename scalar_t, typename func_t>
+void batch_iterator_with_broadcasting(const Tensor& a, const Tensor& b, const func_t& f) {
```

Comment:
That's a very cool function. It really needs some documenting comments to be used by others in the future.